components/camel-bindy/src/test/java/org/apache/camel/dataformat/bindy/csv/BindySimpleCsvFunctionWithClassMethodTest.java

```diff
@@ -0,0 +1,123 @@
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+
+import org.apache.camel.EndpointInject;
+import org.apache.camel.RoutesBuilder;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.dataformat.bindy.annotation.CsvRecord;
+import org.apache.camel.dataformat.bindy.annotation.DataField;
+import org.apache.camel.dataformat.bindy.util.ConverterUtils;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.junit.Test;
+
+public class BindySimpleCsvFunctionWithClassMethodTest extends CamelTestSupport {
+    
+    @EndpointInject(uri = "mock:resultMarshal1")
+    private MockEndpoint mockEndPointMarshal1;
+    
+    @EndpointInject(uri = "mock:resultUnMarshal1")
+    private MockEndpoint mockEndPointUnMarshal1;
+    
+    @EndpointInject(uri = "mock:resultMarshal2")
+    private MockEndpoint mockEndPointMarshal2;
+    
+    @EndpointInject(uri = "mock:resultUnMarshal2")
+    private MockEndpoint mockEndPointUnMarshal2;
+    
+    @Test
+    public void testUnMarshallMessage() throws Exception {
+
+        mockEndPointMarshal1.expectedMessageCount(1);
+        mockEndPointMarshal1.expectedBodiesReceived("\"123\",\"\"\"foo\"\"\",\"10\"" + ConverterUtils.getStringCarriageReturn("WINDOWS"));
+
+        BindyCsvRowFormat7621 body = new BindyCsvRowFormat7621();
+        body.setFirstField("123");
+        body.setSecondField("\"\"foo\"\"");
+        body.setNumber(new BigDecimal(10));
+        template.sendBody("direct:startMarshal1", body);
+        
+        assertMockEndpointsSatisfied();
+        
+        BindyCsvRowFormat7621 model = mockEndPointUnMarshal1.getReceivedExchanges().get(0).getIn().getBody(BindyCsvRowFormat7621.class);
+        
+        assertEquals("123", model.getFirstField());
+        assertEquals("\"\"FOO\"\"", model.getSecondField());
+        assertEquals(new BigDecimal(10), model.getNumber());
+    }
+
+    @Override
+    protected RoutesBuilder createRouteBuilder() throws Exception {
+        return new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                BindyCsvDataFormat camelDataFormat1 =
+                    new BindyCsvDataFormat(BindyCsvRowFormat7621.class);
+                
+                from("direct:startMarshal1")
+                    .marshal(camelDataFormat1)
+                    .to("mock:resultMarshal1")
+                    .to("direct:middle1");
+                
+                from("direct:middle1")
+                    .unmarshal(camelDataFormat1)
+                    .to("mock:resultUnMarshal1");
+            }
+        };
+    }
+
+    @CsvRecord(separator = ",", quote = "\"", quoting = true, quotingEscaped = false)
+    public static class BindyCsvRowFormat7621 implements Serializable {
+
+        @DataField(pos = 1)
+        private String firstField;
+
+        @DataField(pos = 2, function = "toUpperCase")
```

Comment:
Maybe it should be called `method` or `methodCall` instead of `function` as its not really a function, its calling a method. Users may think a function you can do more like `if $it == 2 ? 'two' : 'not two'` etc.

Reply:
Yes, your suggestion is better. Updated.